### DISCUSSION TITLE
Sending the video files & images

### DISCUSSION THREAD

**Discussion by stserakhau** [Auties00/WhatsappWeb4j#209] [Q&A]
Hi guys,

who knows how to send video / image messages ?

I tried these cases, but nothing arrived (text messages is ok, sending document (mp4 as file for download) is ok)

Code which persist thumbnail file
```
BufferedImage img = java2DFrameConverter.convert(frame);
ImageIO.write(img, "jpg", file);
```

Case 1
```
VideoMessage message = VideoMessage.builder()
                        .decodedMedia(FileUtils.readFileToByteArray(file)) // mp4 data
                        .thumbnail(thumbnail) // byte array of thumbnail jpg
                        .caption(fce.getContextPath())  // path like '/a/b/c/d/e/file.mp4'
                        .width(fce.option("width")) // 800
                        .height(fce.option("height")) // 600
                        .duration(fce.option("duration")) // time in milliseconds
                        .mimetype("video/mp4")
                        .build();
ContactJid contact = ...;
this.whatsapp.sendMessage(contact, message);
```

Case 2

```
ImageMessage message = ImageMessage.builder()
                                .caption(caption)
                                .decodedMedia(mediaResult.data)
                                .width(mediaResult.width)
                                .height(mediaResult.height)
                                .mimetype("image/jpg")
                                .build()
ContactJid contact = ...;
this.whatsapp.sendMessage(contact, message);
```

**Comment by Auties00** [comment]
> Hi guys,
> 
> who knows how to send video / image messages ?
> 
> I tried these cases, but nothing arrived (text messages is ok, sending document (mp4 as file for download) is ok)
> 
> Code which persist thumbnail file
> ```
> BufferedImage img = java2DFrameConverter.convert(frame);
> ImageIO.write(img, "jpg", file);
> ```
> 
> Case 1
> ```
> VideoMessage message = VideoMessage.builder()
>                         .decodedMedia(FileUtils.readFileToByteArray(file)) // mp4 data
>                         .thumbnail(thumbnail) // byte array of thumbnail jpg
>                         .caption(fce.getContextPath())  // path like '/a/b/c/d/e/file.mp4'
>                         .width(fce.option("width")) // 800
>                         .height(fce.option("height")) // 600
>                         .duration(fce.option("duration")) // time in milliseconds
>                         .mimetype("video/mp4")
>                         .build();
> ContactJid contact = ...;
> this.whatsapp.sendMessage(contact, message);
> ```
> 
> Case 2
> 
> ```
> ImageMessage message = ImageMessage.builder()
>                                 .caption(caption)
>                                 .decodedMedia(mediaResult.data)
>                                 .width(mediaResult.width)
>                                 .height(mediaResult.height)
>                                 .mimetype("image/jpg")
>                                 .build()
> ContactJid contact = ...;
> this.whatsapp.sendMessage(contact, message);
> ```

What version are you using?

**Reply by stserakhau** [reply]
3.0.5
```
        <dependency>
            <groupId>com.github.auties00</groupId>
            <artifactId>whatsappweb4j</artifactId>
            <version>3.0.5</version>
        </dependency>
```

**Reply by Auties00** [reply]
Can you send a case that doesn't work and the media you are trying to send so I can try?

**Reply by Auties00** [reply]
Turns out it was a bug, will be fixed in 3.1.1

**Reply by stserakhau** [reply]
thanks !!!
